### ISSUE TITLE
Additional Audio input problems with macOS Big Sur

### ISSUE THREAD

**Issue by paulcanning** [benfry/processing4#182]
Downloaded this version as audio permissions are broken on Big Sur, and are still broken on Big Sur. No prompt to ask for access or anything.

**Comment by kevinstadler** [comment]
This looks like it's the Big Sur iteration of the earlier Catalina/Mojave not asking for microphone/camera capture permission problem (see processing/processing-sound#51 and processing/processing-video#134) that was fixed at the Processing level with c5baab0 (release 4.0a3).

In the past month separate people have reported capture failing on Big Sur for both video (processing/processing-video#178) and audio (processing/processing-sound#64), but again this will probably have to be fixed on the Processing app-bundling level, possibly another change to the `.plist` file? I haven't been able to find any information on any new permission keys introduced with Big Sur though...

### Potential (temporary) workaround

https://github.com/processing/processing-video/issues/178#issuecomment-800529065 appears to have managed to force Processing to be added to the list of apps in the OSX System Preferences > Permission pane by running some shell commands and starting Processing from the shell (only required once): http://wiki.iad.zhdk.ch/CV/904626177/Video+Library+2.0

The short version:
```
# tell OSX to de-quarantine the sound and/or video library
xattr -r -d com.apple.quarantine ~/Documents/Processing/libraries/sound/*
xattr -r -d com.apple.quarantine ~/Documents/Processing/libraries/video/*
# start processing
/Applications/Processing.app/Contents/MacOS/Processing
# now RUN A SKETCH that requires audio/video input, and grant capturing permission when asked
```

OSX stores this app permission for all later runs of Processing (including those not run from the shell)

**Comment by benfry** [comment]
Video and Audio seem to be having similar problems. 

Try with this version, it adds more entitlements for the macOS “hardened runtime” setup, which may be helpful. https://download.processing.org/processing-1274-macosx.zip

But even without those, I can get video (and I presume audio) capture working on Big Sur. I need more involvement from the authors of those libraries to debug the problem. 

With 4.0 alpha 4 (as well as the linked version here), audio input is working fine on the first try. I get the prompt:

<img width="578" alt="Screen Shot 2021-06-18 at 3 50 33 PM" src="https://user-images.githubusercontent.com/1623101/122610172-3084c980-d04d-11eb-9d0e-87e4d2528dfa.png">

And then making noise shows large circles on the screen.

That's on an M1 laptop running a fully up-to-date Big Sur release, downloading the software from processing.org and running it. Not a development version.

If you're not getting the prompt, you can reset it with `tccutil reset Microphone` (or `Camera` if you're attempting with video).

**Comment by benfry** [comment]
Closing for now; can't reproduce any issues from the PDE side of things. (Not that it's impossible that there's more to add to the PDE, but I also can't chase down issues with the video and sound libraries in addition to everything else.)

Entitlements tweak is here: https://github.com/processing/processing4/commit/7b75acf2799f61c9c22233f38ee73c07635cea14
